packages/govuk-frontend/src/govuk/overrides/_typography.scss

```
@@ -3,9 +3,22 @@
 
   // Generate typography override classes for each responsive font map in the
   // typography scale eg .govuk-\!-font-size-80
-  @each $size in map-keys($govuk-typography-scale) {
-    .govuk-\!-font-size-#{$size} {
-      @include govuk-font-size($size, $important: true);
+  //
+  // govuk-!-font-size-14 is deprecated
+  @each $size, $font-map in $govuk-typography-scale {
+    // suppress class if the map key is a string eg: _14 to avoid classes like
+    // govuk-!-font-size-_14 getting generated
+    @if type-of($size) == "number" {
```

Sorry @owenatgov, but what if a fork of GOV.UK Frontend has switched to t-shirt sizing?

This change will stop their `"string"` key names from working

How about we still allow strings but only if the first character is NOT an underscore?

```suggestion
    @if type-of($size) == "number" or (type-of($size) == "string" and str-slice($size, 1, 1) != "_") {
```

---

I've found a way to avoid the duplicate `_14` entry in `$govuk-typography-scale`

I've changed all the keys to strings but updating `$size` to `#{$size}` in our utilities allows that

Probably considered a breaking change but was a nice exercise to try:
https://github.com/alphagov/govuk-frontend/compare/deprecate-14...deprecate-14-strings

---

Sorry @owenatgov I must have [changed my suggestion above](https://github.com/alphagov/govuk-frontend/pull/4649#discussion_r1474385602) too late

I remembered we don't need `type-of()` if we use `#{$size}` to let Sass do the interpolation

---

Captured the update! I'm not sure if that string fix would be a breaking change or not. It's really cool but would it be ok if we tackled it separately to this?

---

Of course. I've split it out into another PR:

* https://github.com/alphagov/govuk-frontend/pull/4713